tests/DataProvider/ChainCollectionDataProviderTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Tests\DataProvider;
+
+use ApiPlatform\Core\DataProvider\ChainCollectionDataProvider;
+use ApiPlatform\Core\DataProvider\CollectionDataProviderInterface;
+use ApiPlatform\Core\Exception\ResourceClassNotSupportedException;
+use ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\Dummy;
+
+/**
+ * Retrieves items from a persistence layer.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ChainCollectionDataProviderTest extends \PHPUnit_Framework_TestCase
+{
+    public function testGetCollection()
+    {
+        $dummy = new Dummy();
+        $dummy->setName('Rosa');
+        $dummy2 = new Dummy();
+        $dummy2->setName('Parks');
+
+        $firstDataProvider = $this->prophesize(CollectionDataProviderInterface::class);
+        $firstDataProvider->getCollection(Dummy::class, null)->willReturn([$dummy, $dummy2]);
+
+
+        $dataProviders = [$firstDataProvider->reveal()];
```

Comment:
It would be nice to add a data provider returning null first to test that the second is executed. Same apply everywhere.